### PR TITLE
Cmr 326

### PR DESCRIPTION
**Pull request by ruegamer**
@claussni this includes:

- correct DOI-Number if present
- roleTerms of persons and corporations (correct author, degree granting institution as second, editor as third option, if no publisher is given)

### CODE REVIEW

File: Classes/Helper/DataCiteXml.php [hunk 1]
```diff
@@ -33,7 +33,7 @@ public static function convertFromMetsXml($metsXml)
         $metsXml->registerXPathNamespace('slub', 'http://slub-dresden.de/');
 
         // doi
-        $metsDoi = $metsXml->xpath("//mods:identifier[@type='qucosa:doi']");
```
Comment (claussni) [inline Classes/Helper/DataCiteXml.php]: What was the reason we used qucosa:doi before? And why did this change?

Comment (ruegamer) [reply Classes/Helper/DataCiteXml.php]: I'm sorry, but I can't remember.

File: Classes/Helper/DataCiteXml.php [hunk 2]
```diff
@@ -46,6 +46,10 @@ public static function convertFromMetsXml($metsXml)
         foreach($metsCreator as $creator)
         {
             $creator->registerXPathNamespace('mods', 'http://www.loc.gov/mods/v3');
+            $role = $creator->xpath(".//mods:roleTerm[@type='code']");
```
Comment (claussni) [inline Classes/Helper/DataCiteXml.php]: The creators element is always empty:

```
<?xml version="1.0" encoding="UTF-8"?>
<resource xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://datacite.org/schema/kernel-4" xsi:schemaLocation="http://datacite.org/schema/kernel-4 http://schema.datacite.org/meta/kernel-4/metadata.xsd">
  <identifier identifierType="DOI">doi:foo/123</identifier>
  <creators/>
  <titles>
    <title>Test Dissertation</title>
  </titles>
  <publisher>TU Dresden</publisher>
  <publicationYear>2017</publicationYear>
  <subjects>
    <subject>xxx</subject>
  </subjects>
  <language>de</language>
  <resourceType resourceTypeGeneral="Text">doctoral_thesis</resourceType>
</resource>

```

Comment (ruegamer) [reply Classes/Helper/DataCiteXml.php]: There was a wrong condition for authors/composers. See commit [5245bce](https://github.com/kitodo/kitodo-publication/pull/70/commits/5245bce1717db1620865b52d60aa98cfeee2ca41).

**Pull request merged**